portmapper/mapper.go

```diff
@@ -122,35 +124,32 @@ func (pm *PortMapper) MapRange(container net.Addr, hostIP net.IP, hostPortStart,
 		}
 	}()
 
+	pm.lock.Lock()
```

Comment:
Let's not hold the lock around unnecessary code. It just needs to be protect `currentMappings`. in fact this code be rewritten like this:
```
key := getKey(m.host)

pm.lock.Lock()
_, exists := pm.currentMappings[key]
pm.lock.Unlock()

if exists {
     return nil, ErrPortMappedForIP
}
```